### PR TITLE
feat(ui_auth): added Hungarian localization

### LINKED ISSUE #9054: [ui_auth, ui_firestore, ui_database] Add Hungarian localization to firebase_ui
### Discussed in https://github.com/firebase/flutterfire/discussions/9052

<div type='discussions-op-text'>

<sup>Originally posted by **cselti** July  7, 2022</sup>
Add Hungarian localization flutterfire_ui.</div>

### PR DESCRIPTION
**Pull request by koltai75**
## Description

Added Hungarian localization to Firebase UI Auth

## Related issues

Closes #9054

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Review by lesnitsky — APPROVED** [review]

**Comment by koltai75** [comment]
@lesnitsky I also found out that some labels are hardcoded, for example [here](https://github.com/firebase/flutterfire/blob/dd611cd25139f78bc947a68366213902a31ae350/packages/firebase_ui_auth/lib/src/screens/email_verification_screen.dart#L178) and [here](https://github.com/firebase/flutterfire/blob/dd611cd25139f78bc947a68366213902a31ae350/packages/firebase_ui_auth/lib/src/screens/email_verification_screen.dart#L199) in the email verification screen. I experimented a little and added some new labels but it turned out that it does not handle if new label definitions are missing from existing localizations. I could create another PR for that, but what is your suggestion, how to handle the different localizations? Should I add these in English to every arb?

**Comment by lesnitsky** [comment]
@koltai75 no need to assign, @Kukkerem feel free to review this PR

as of hardcoded labels – we're looking into adding translations to those early next year, so no action required from you.

**Comment by lesnitsky** [comment]
@koltai75, seems like @Kukkerem is not responsive, do you know someone else who speaks Hungarian and could review this PR?

File: packages/firebase_ui_localizations/lib/src/lang/hu.dart [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+  String get linkEmailButtonText => "Következő";
+
+  @override
+  String get longitudeLabel => "hosszúság";
+
+  @override
+  String get mapLabel => "térkép";
+
+  @override
+  String get mfaTitle => "2 lépcsős ellenőrzés";
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/src/lang/hu.dart:118]: Kétlépcsős/Kétlépéses azonosítás/ellenőrzés/hitelesítés

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+  "@longitudeLabel": {
+    "description": "",
+    "placeholders": {}
+  },
+  "mapLabel": "térkép",
+  "@mapLabel": {
+    "description": "",
+    "placeholders": {}
+  },
+  "mfaTitle": "2 lépcsős ellenőrzés",
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb:178]: Kétlépcsős/Kétlépéses azonosítás/ellenőrzés/hitelesítés

File: packages/firebase_ui_localizations/lib/src/lang/hu.dart [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+  @override
+  String get name => "Név";
+
+  @override
+  String get northInitialLabel => "É";
+
+  @override
+  String get nullLabel => "nulla";
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/src/lang/hu.dart:127]: Ez lehetne inkább "null érték".

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+  "@name": {
+    "description": "Used as a placeholder of the EditableUserDisplayName.",
+    "placeholders": {}
+  },
+  "northInitialLabel": "É",
+  "@northInitialLabel": {
+    "description": "",
+    "placeholders": {}
+  },
+  "nullLabel": "nulla",
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb:193]: Ez lehetne inkább "null érték".

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+  "@passwordInputLabel": {
+    "description": "Used as a label of the PasswordInput.",
+    "placeholders": {}
+  },
+  "passwordIsRequiredErrorText": "A jelszó kötelező",
+  "@passwordIsRequiredErrorText": {
+    "description": "Used as an error text when PasswordInput is empty.",
+    "placeholders": {}
+  },
+  "passwordResetEmailSentText": "Küldtünk egy levelet a jelszó visszaállításához. Kérjük, ellenőrizze a postaládáját!",
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb:223]: Levelet helyett talán "e-mailt"?

File: packages/firebase_ui_localizations/lib/src/lang/hu.dart [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+
+  @override
+  String get signInText => "Belépés";
+
+  @override
+  String get signInWithAppleButtonText => "Apple belépés";
+
+  @override
+  String get signInWithEmailLinkSentText =>
+      "Küldtünk egy levelet a varázs hivatkozással. Kérjük, ellenőrizze a postaládáját és kattintson a hivatkozásra a belépéshez!";
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/src/lang/hu.dart:201]: Levelet helyett talán "e-mailt"? A "Belépési link" jobban hangzana varázs hivatkozás helyett.

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb [hunk 7]
```diff
@@ -0,0 +1,427 @@
...
+  "@signInText": {
+    "description": "Used as a title of the LoginView when AuthAction is AuthAction.signIn.",
+    "placeholders": {}
+  },
+  "signInWithAppleButtonText": "Apple belépés",
+  "@signInWithAppleButtonText": {
+    "description": "Used as a label of the OAuthProviderButton for Apple provider.",
+    "placeholders": {}
+  },
+  "signInWithEmailLinkSentText": "Küldtünk egy levelet a varázs hivatkozással. Kérjük, ellenőrizze a postaládáját és kattintson a hivatkozásra a belépéshez!",
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb:317]: Levelet helyett talán "e-mailt"? A "Belépési link" jobban hangzana varázs hivatkozás helyett.

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb [hunk 8]
```diff
@@ -0,0 +1,427 @@
...
+  "resetPasswordButtonLabel": "Jelszó visszaállítása",
+  "@resetPasswordButtonLabel": {
+    "description": "Used as a label of submit button of the ForgotPasswordView.",
+    "placeholders": {}
+  },
+  "@runtimeType": {
+    "description": "A representation of the runtime type of the object.",
+    "placeholders": {}
+  },
+  "sendLinkButtonLabel": "Varázs hivatkozás küldése",
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb:287]: A "Belépési link" jobban hangzana.

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb [hunk 9]
```diff
@@ -0,0 +1,427 @@
...
+  "@signInWithAppleButtonText": {
+    "description": "Used as a label of the OAuthProviderButton for Apple provider.",
+    "placeholders": {}
+  },
+  "signInWithEmailLinkSentText": "Küldtünk egy levelet a varázs hivatkozással. Kérjük, ellenőrizze a postaládáját és kattintson a hivatkozásra a belépéshez!",
+  "@signInWithEmailLinkSentText": {
+    "description": "Indicates that email with a sign in link was sent.",
+    "placeholders": {}
+  },
+  "signInWithEmailLinkViewTitleText": "Belépés varázs hivatkozással",
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb:322]: A "Belépési link" jobban hangzana varázs hivatkozás helyett.

File: packages/firebase_ui_localizations/lib/src/lang/hu.dart [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+  String get eastInitialLabel => "K";
+
+  @override
+  String get emailInputLabel => "E-mail";
+
+  @override
+  String get emailIsRequiredErrorText => "Az e-mail megadása kötelező";
+
+  @override
+  String get emailLinkSignInButtonLabel => "Belépés varázs hivatkozással";
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/src/lang/hu.dart:65]: A "Belépési link" jobban hangzana varázs hivatkozás helyett.

File: packages/firebase_ui_localizations/lib/src/lang/hu.dart [hunk 11]
```diff
@@ -0,0 +1,267 @@
...
+  String get registerHintText => "Nincs még fiókja?";
+
+  @override
+  String get registerText => "Regisztráció";
+
+  @override
+  String get resetPasswordButtonLabel => "Jelszó visszaállítása";
+
+  @override
+  String get sendLinkButtonLabel => "Varázs hivatkozás küldése";
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/src/lang/hu.dart:182]: A "Belépési link" jobban hangzana varázs hivatkozás helyett.

File: packages/firebase_ui_localizations/lib/src/lang/hu.dart [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+
+  @override
+  String get signInWithAppleButtonText => "Apple belépés";
+
+  @override
+  String get signInWithEmailLinkSentText =>
+      "Küldtünk egy levelet a varázs hivatkozással. Kérjük, ellenőrizze a postaládáját és kattintson a hivatkozásra a belépéshez!";
+
+  @override
+  String get signInWithEmailLinkViewTitleText => "Belépés varázs hivatkozással";
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/src/lang/hu.dart:204]: A "Belépési link" jobban hangzana varázs hivatkozás helyett.

File: packages/firebase_ui_localizations/lib/src/lang/hu.dart [hunk 13]
```diff
@@ -0,0 +1,267 @@
...
+
+  @override
+  String get passwordInputLabel => "Jelszó";
+
+  @override
+  String get passwordIsRequiredErrorText => "A jelszó kötelező";
+
+  @override
+  String get passwordResetEmailSentText =>
+      "Küldtünk egy levelet a jelszó visszaállításához. Kérjük, ellenőrizze a postaládáját!";
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/src/lang/hu.dart:146]: Levelet helyett talán "e-mailt"?

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb [hunk 14]
```diff
@@ -0,0 +1,427 @@
...
+  "@emailInputLabel": {
+    "description": "Used as a label of the EmailInput.",
+    "placeholders": {}
+  },
+  "emailIsRequiredErrorText": "Az e-mail megadása kötelező",
+  "@emailIsRequiredErrorText": {
+    "description": "Used as an error text of the EmailInput when the email is empty.",
+    "placeholders": {}
+  },
+  "emailLinkSignInButtonLabel": "Belépés varázs hivatkozással",
```
Comment (Kukkerem) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_hu.arb:89]: A "Belépési link" jobban hangzana varázs hivatkozás helyett.

**Review by Kukkerem — CHANGES_REQUESTED** [review]
I have only found a few small things, but all in all, excellent work!

**Comment by koltai75** [comment]
@lesnitsky I changed some localization texts according to the suggestions of @Kukkerem.

**Review by Kukkerem — APPROVED** [review]
It seems perfect!

**Review by pr-Mais — APPROVED** [review]

**Review by Lyokone — APPROVED** [review]

**Pull request merged**